tests/tests_integration/test_api/test_hosted_extractors/test_jobs.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.hosted_extractors import (
+    CogniteFormat,
+    Destination,
+    Job,
+    JobList,
+    JobLogsList,
+    JobMetricsList,
+    JobUpdate,
+    JobWrite,
+    Source,
+)
+from cognite.client.exceptions import CogniteAPIError
+from cognite.client.utils._text import random_string
+
+# @pytest.fixture
+# def one_source(cognite_client: CogniteClient) -> Source:
+#     my_source = EventHubSourceWrite(
+#         external_id=f"myNewSource-{random_string(10)}",
+#         host="myhost",
+#         event_hub_name="myeventhub",
+#         key_name="mykeyname",
+#         key_value="secret",
+#     )
+#     retrieved = cognite_client.hosted_extractors.sources.retrieve(my_source.external_id, ignore_unknown_ids=True)
+#     if retrieved:
+#         yield retrieved
+#     created = cognite_client.hosted_extractors.sources.create(my_source)
+#     yield created
+#
+#     cognite_client.hosted_extractors.sources.delete(created.external_id, ignore_unknown_ids=True)
+
+
+@pytest.fixture
+def one_job(cognite_client: CogniteClient, one_event_hub_source: Source, one_destination: Destination) -> Job:
+    my_job = JobWrite(
+        external_id=f"myJobForTesting-{random_string(10)}",
+        destination_id=one_destination.external_id,
+        source_id=one_event_hub_source.external_id,
+        format=CogniteFormat(encoding="utf16"),
+    )
+    retrieved = cognite_client.hosted_extractors.jobs.retrieve(my_job.external_id, ignore_unknown_ids=True)
+    if retrieved:
+        yield retrieved
+    created = cognite_client.hosted_extractors.jobs.create(my_job)
+    yield created
+
+    cognite_client.hosted_extractors.jobs.delete(created.external_id, ignore_unknown_ids=True)
+
+
+class TestJobs:
+    def test_create_update_retrieve_delete(
+        self, cognite_client: CogniteClient, one_destination: Destination, one_event_hub_source: Source
+    ) -> None:
+        my_job = JobWrite(
+            external_id=f"myJobForTesting-{random_string(10)}",
+            destination_id=one_destination.external_id,
+            source_id=one_event_hub_source.external_id,
+            format=CogniteFormat(
+                encoding="utf16",
+            ),
+        )
+        created: Job | None = None
+        try:
+            created = cognite_client.hosted_extractors.jobs.create(my_job)
+            assert isinstance(created, Job)
+            update = JobUpdate(external_id=my_job.external_id).format.set(CogniteFormat(encoding="utf16le"))
+            updated = cognite_client.hosted_extractors.jobs.update(update)
+            format_ = updated.format
+            assert isinstance(format_, CogniteFormat)
+            assert format_.encoding == "utf16le"
```

Comment:
I think this reads better but up to you!
```suggestion
            assert isinstance(updated.format, CogniteFormat)
            assert updated.format.encoding == "utf16le"
```